apis/r/R/SOMADataFrame.R

```diff
@@ -327,9 +327,65 @@ SOMADataFrame <- R6::R6Class(
     #' @return None, instead a \code{\link{.NotYetImplemented}()} error is raised
     #'
     shape = function() stop(errorCondition(
-      "'SOMADataFrame$shape()' is not implemented yet",
+      "'SOMADataFrame$shape()' is not implemented",
```

Comment:
I had purposefully added the "yet" to adhere to R's existing [`.NotYetImplemented()`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/notyet.html)
```r
> .NotYetImplemented()
Error: '.NotYetImplemented' is not implemented yet
```

Reply:
OK, I'll put it back

(Sounds like a spec discussion ...)

I would like it if `shape` could return what `.maybe_soma_joinid_shape` does -- remembering that a `SOMADataFrame` per the spec must have a dim or attr named `soma_joinid` but (a) it doesn't have to be the only dim, and (b) it doesn't even have to be dim at all. (This is context.)

Reply:
(Also, Python just raised `NotImplementedError`, not a `NotImplementedYetError` -- which is a difference between the two, albeit minor.)

Reply:
Done on this commit: https://github.com/single-cell-data/TileDB-SOMA/pull/2953/commits/840ed2dbd355a6759cc2322ea1e4e2ac37c0c5e6

Reply:
> I personally would love it if `shape` could return what `.maybe_soma_joinid_shape` [...]

That would require re-hashing the discussions that lead to #2533 and #2531; I'm happy to have those conversations and merge these behaviors, but I agree that this is a "task for later"

> (Also, Python just raises `NotImplementedError`, not a `NotImplementedYetError` [...]

I'm aware, this is R's equivalent. I get that these are different, but whenever possible I'd like to adhere to existing R standards

> Done on this commit: 840ed2d

TY!